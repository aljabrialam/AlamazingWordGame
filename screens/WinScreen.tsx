import React from 'react';

import {Text, Share, View, Alert, TouchableOpacity} from 'react-native';
import styles from '../styles';
interface IProps {
  navigation: any;
}

export default class WinScreen extends React.Component<IProps> {
  static navigationOptions = {
    title: '',
    headerStyle: {
      elevation: 0,
      shadowOpacity: 0,
      borderBottomWidth: 0,
    },
    headerLeft: () => <View />,
  };

  goBack() {
    const {navigation} = this.props;
    navigation.goBack();
  }

  render() {
    const score = this.props.navigation.getParam('score', '');
    // const score = this.props.navigation.score
    const onShare = async () => {
      try {
        const result = await Share.share({
          message:
            'Singtel Word Game App - Awesome I earned ' + score + ' points!',
        });
        if (result.action === Share.sharedAction) {
          if (result.activityType) {
            // shared with activity type of result.activityType
          } else {
            // shared
          }
        } else if (result.action === Share.dismissedAction) {
          // dismissed
        }
      } catch (error: any) {
        Alert.alert(error.message);
      }
    };

    return (
      <View style={styles.homeContainer}>
        <Text style={[styles.pageCountText, {marginBottom: 200}]}>1/3</Text>

        <Text style={styles.winText}>Correct!</Text>
        <Text style={styles.winText}>Congratulations</Text>
        <Text style={[styles.winText, {marginBottom: 100}]}>
          You earn {score} points
        </Text>

        <View style={styles.startButtonContainer}>
          <TouchableOpacity
            hitSlop={{top: 30, bottom: 30, left: 30, right: 30}}
            style={styles.gameTitleView}
            onPress={() => this.goBack()}>
            <Text style={{marginTop: 8, fontSize: 20}}>NEXT</Text>
          </TouchableOpacity>
        </View>

        <View style={[styles.startButtonContainer, {marginTop: 20}]}>
          <TouchableOpacity
            hitSlop={{top: 30, bottom: 30, left: 30, right: 30}}
            style={styles.gameTitleView}
            onPress={() => onShare()}>
            <Text style={{marginTop: 8, fontSize: 20}}>SHARE MY SCORE!</Text>
          </TouchableOpacity>
        </View>
      </View>
    );
  }
}
