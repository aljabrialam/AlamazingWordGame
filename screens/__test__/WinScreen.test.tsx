import React from 'react';
import renderer from 'react-test-renderer';

import WinScreen from '../WinScreen';

const navigationPropsLanding = {
  navigation: {
    navigate: jest.fn(),
    addListener: jest.fn(),
  },
  route: {params: {score: 0}},
};

describe('<WinScreen />', () => {
  it('has 5 child', () => {
    const tree: any = renderer
      .create(<WinScreen {...navigationPropsLanding} />)
      .toJSON();
    expect(tree.children.length).toBe(5);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<WinScreen {...navigationPropsLanding} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
