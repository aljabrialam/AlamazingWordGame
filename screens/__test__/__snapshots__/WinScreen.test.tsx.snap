// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WinScreen /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      [
        {
          "fontSize": 20,
          "fontWeight": "500",
          "justifyContent": "flex-start",
          "padding": 10,
          "textAlign": "center",
          "width": "100%",
        },
        {
          "marginBottom": 200,
        },
      ]
    }
  >
    1/3
  </Text>
  <Text
    style={
      {
        "fontSize": 30,
        "fontWeight": "500",
        "padding": 5,
        "textAlign": "center",
        "width": "100%",
      }
    }
  >
    Correct!
  </Text>
  <Text
    style={
      {
        "fontSize": 30,
        "fontWeight": "500",
        "padding": 5,
        "textAlign": "center",
        "width": "100%",
      }
    }
  >
    Congratulations
  </Text>
  <Text
    style={
      [
        {
          "fontSize": 30,
          "fontWeight": "500",
          "padding": 5,
          "textAlign": "center",
          "width": "100%",
        },
        {
          "marginBottom": 100,
        },
      ]
    }
  >
    You earn 
     points
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FECBCA",
        "borderColor": "#d3d6da",
        "borderWidth": 2,
        "height": 50,
        "marginTop": 50,
        "paddingEnd": 30,
        "paddingStart": 30,
        "width": 240,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={
        {
          "bottom": 30,
          "left": 30,
          "right": 30,
          "top": 30,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "flexDirection": "row",
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 20,
            "marginTop": 8,
          }
        }
      >
        NEXT
      </Text>
    </View>
  </View>
</View>
`;
